selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java

```diff
@@ -1967,6 +1968,64 @@ public int getMarkersQuantity(MarkerLocator markerLocator) {
         .size();
   }
 
+  /**
+   * Waits {@code quantity} of specified {@code markerLocator} during {@code timeout}.
+   *
+   * @param markerLocator type of the expected markers
+   * @param quantity expected quantity of the markers
+   * @param timeout time in seconds for waiting expected quantity of the markers
+   */
+  public void waitMarkersQuantity(MarkerLocator markerLocator, int quantity, int timeout) {
+    seleniumWebDriverHelper.waitSuccessCondition(
+        driver -> quantity == getMarkersQuantity(markerLocator), timeout);
+  }
+
+  /**
+   * Waits {@code quantity} of specified {@code markerLocator}.
+   *
+   * @param markerLocator type of the expected markers
+   * @param quantity expected quantity of the markers
+   */
+  public void waitMarkersQuantity(MarkerLocator markerLocator, int quantity) {
+    waitMarkersQuantity(markerLocator, quantity, LOAD_PAGE_TIMEOUT_SEC);
+  }
+
+  /**
+   * Waits quantity of specified {@code markerLocator} between {@code lowerBound} and {@code
+   * upperBound} values during {@code timeout}.
+   *
+   * @param markerLocator type of the expected markers
+   * @param lowerBound the lower bound of the markers quantity
+   * @param upperBound the upper bound of the markers quantity
+   * @param timeout time in seconds for waiting expected quantity of the markers
+   */
+  public void waitMarkersQuantityBetween(
+      MarkerLocator markerLocator, int lowerBound, int upperBound, int timeout) {
+    seleniumWebDriverHelper.waitSuccessCondition(
+        driver -> {
+          final int markersQuantity = getMarkersQuantity(markerLocator);
+
+          final boolean isQuantityInLowerBound = lowerBound <= markersQuantity;
+          final boolean isQuantityInUpperBound = upperBound >= markersQuantity;
+
+          return isQuantityInLowerBound && isQuantityInUpperBound;
```

Comment:
Looks over-complicated, IMHO.
Also there is lack of information about bounds' inclusiveness.